server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -3265,6 +3265,11 @@ private EngineConfig newEngineConfig(LongSupplier globalCheckpointSupplier) {
             }
         };
         final boolean isTimeBasedIndex = mapperService == null ? false : mapperService.mappingLookup().hasTimestampField();
+        if (mapperService != null) {
+            // we build analyzers that require longer running operations and blocking calls
+            // here during shard recovery in the generic thread pool.
+            mapperService.reloadSearchAnalyzers();
```

Comment:
We are holding various mutexes when we get here, and we also get here on certain primary failovers, so this feels like the wrong place to do this. But admittedly I don't know what the exact dependencies are. Can we move it earlier (or later) in the recovery process? For instance could we do this in `org.elasticsearch.index.shard.IndexEventListener#beforeIndexShardRecovery` (which is already a naturally async step of the recovery process)?

Reply:
@DaveCTurner  Thanks for the feedback. Addressed in https://github.com/elastic/elasticsearch/pull/96114/commits/bb367f9fbe385db4ca0ae033e6fd35f49c95abfd